File: targets/python-web/vulns/vuln40_session.py
from flask import Blueprint, request, make_response, jsonify
bp = Blueprint("vuln40", __name__)

@bp.get("/vuln40/setsession")
def setsession():
    sid = request.args.get("sid", "")
    resp = make_response(jsonify({"session": sid}))
    # VULN:VULN-40:CWE-384:logic session id taken from the request and set as-is
    resp.set_cookie("session", sid)
    return resp

File: targets/jsts-web/src/vulns/vuln40_session.ts
import { Router } from "express";
export const router = Router();
router.get("/vuln40/setsession", (req, res) => {
  const sid = String(req.query.sid || "");
  // VULN:VULN-40:CWE-384:logic session id taken from the request and set as-is
  res.cookie("session", sid);
  res.json({ session: sid });
});
